crypto/evp/evp_fetch.c

```diff
@@ -66,24 +65,45 @@ static OSSL_METHOD_STORE *get_default_method_store(OPENSSL_CTX *libctx)
                                 &default_method_store_method);
 }
 
+/*
+ * To identity the method in the method store, we mix the name identity
+ * with the operation identity, with the assumption that we don't have
+ * more than 2^24 names or more than 2^8 operation types.
+ *
+ * The resulting identity is a 32-bit integer, composed like this:
+ *
+ * +---------24 bits--------+-8 bits-+
+ * |      name identity     | op id  |
+ * +------------------------+--------+
+ */
+static uint32_t method_id(unsigned int operation_id, unsigned int name_id)
+{
+    if (!ossl_assert(name_id < (1 << 24) || operation_id < (1 << 8))
+        || !ossl_assert(name_id > 0 && operation_id > 0))
+        return 0;
+    return ((name_id << 8) & 0xFFFFFF00) | (operation_id & 0x000000FF);
+}
+
 static void *get_method_from_store(OPENSSL_CTX *libctx, void *store,
-                                   const char *name, const char *propquery,
-                                   void *data)
+                                   int operation_id, const char *name,
+                                   const char *propquery, void *data)
 {
     struct method_data_st *methdata = data;
     void *method = NULL;
     OSSL_NAMEMAP *namemap;
-    int id;
+    int nameid;
+    uint32_t methid;
 
     if (store == NULL
         && (store = get_default_method_store(libctx)) == NULL)
         return NULL;
 
     if ((namemap = ossl_namemap_stored(libctx)) == NULL
-        || (id = ossl_namemap_add(namemap, name)) == 0)
+        || (nameid = ossl_namemap_add(namemap, name)) == 0)
         return NULL;
 
-    (void)ossl_method_store_fetch(store, id, propquery, &method);
+    methid = method_id(operation_id, nameid);
```

Comment:
Don't we need to check `methid` to see if it is 0 and treat it as an error if so?

Reply:
No need. nameid can't be zero here

Reply:
But if name_id > 2^24 or operation_id > 2^8 then it will fail.

Reply:
Ah

Reply:
Fixed